### ISSUE TITLE
Video stream is not centered in the window

### ISSUE THREAD

**Issue by deoxal** [Genymobile/scrcpy#5199]
The mirrored screen is displayed on the edges of the window. It the position varies depending the app. Ideally I would like to resize the mirrored screen and potentially stretch it so it fits my entire screen.

![Screenshot 2024-08-14_23-40-29](https://github.com/user-attachments/assets/607db3d2-99c2-4d1f-8656-ccd57097c92e)

 - **OS:** [Mint 21.3]
 - **Scrcpy version:** [2.6.1]
 - **Installation method:** [manual build]
 - **Device model:** Pixel 8a
 - **Android version:** [14]

This is amazing software though. I've wanted to do this for years but couldn't until now.

**Comment by rom1v** [comment]
One pair of black bars is added by your phone to match the phone aspect ratio. Scrcpy mirrors the whole phone screen (including these bars), but the result does not match your computer screen, so it adds black bars.

To avoid the problem, one solution could be to crop mirroring with `--crop=…`, unfortunately this is broken since Android 14: #4011.

The other solution is to change your physical device screen resolution to match the aspect ratio of your computer display (`adb shell wm size …`).

**Comment by deoxal** [comment]
Is the cropping being done on the phone with that method? Why can't the video be cropped by my PC?

Also this doesn't happen with every app. I'm playing Space Marshals and it happens with 1 but not 2 or 3. I haven't found any other app that does it so far.

**Comment by rom1v** [comment]
> Is the cropping being done on the phone with that method?

Are there any black bars on your phone?

Please run `scrcpy` but keep the window windowed (not maximized). How is the content? Please post a screenshot.

**Comment by deoxal** [comment]
Yes I measured it out by showing the pointer location setting in developer options. A width of 120 pixels are used for a black bar to mask the camera being under the display so just a single bar not a pair. And shrinking the window causes the black bar to scale down with the rest of the display.
